components/brave_wallet/browser/swap_service.cc

```diff
@@ -448,15 +537,29 @@ void SwapService::OnGetJupiterQuote(GetJupiterQuoteCallback callback,
 void SwapService::GetJupiterSwapTransactions(
     mojom::JupiterSwapParamsPtr params,
     GetJupiterSwapTransactionsCallback callback) {
-  if (!IsSolanaNetworkSupported(json_rpc_service_->GetChainIdSync(
-          mojom::CoinType::SOL, absl::nullopt))) {
+  const auto& chain_id =
+      json_rpc_service_->GetChainIdSync(mojom::CoinType::SOL, absl::nullopt);
+  if (!IsSolanaNetworkSupported(chain_id)) {
     std::move(callback).Run(
         nullptr, nullptr,
         l10n_util::GetStringUTF8(IDS_BRAVE_WALLET_UNSUPPORTED_NETWORK));
     return;
   }
 
-  auto encoded_params = EncodeJupiterTransactionParams(std::move(params));
+  auto brave_swap_fee_params = brave_wallet::mojom::BraveSwapFeeParams::New();
+  brave_swap_fee_params->chain_id = chain_id;
+  brave_swap_fee_params->taker = params->user_public_key;
+  brave_swap_fee_params->input_token = params->input_mint;
+  brave_swap_fee_params->output_token = params->output_mint;
+  const auto& brave_fee = GetBraveFeeInternal(std::move(brave_swap_fee_params));
+  double effective_fee_pct;
+  bool has_fee =
+      brave_fee &&
+      base::StringToDouble(brave_fee->effective_fee_pct, &effective_fee_pct) &&
+      effective_fee_pct > 0;
```

Comment:
It seems `has_fee` flag should be a part of `BraveSwapFeeResponse` and be calculated in `GetBraveFeeInternal`

Reply:
Fixed here: https://github.com/brave/brave-core/pull/19834/commits/0eed8dc23e53066c409f28b9cf4ebdb37b206405